lib/QuickBaseClient.rb

```diff
@@ -2324,7 +2337,7 @@ def findDBByname( dbname )
       xmlRequestData = toXML( :dbname, @dbname )
 
       sendRequest( :GrantedDBs, xmlRequestData )
-      @dbid = getResponsePathValue( 'databases/dbinfo/dbid[1]' )
+      @dbid = getResponsePathValueByDBName( 'databases', dbname )

```

Comment:
previously  path value for GrantedDBs API response was hardcoded for path databases/dbinfo/dbid[1] .
It is replaced as above and path changes depending on DB name for which we are requesting